.github/workflows/workflows.yaml

```diff
@@ -0,0 +1,64 @@
+name: Test Suite
+on:
+  pull_request:
+
+jobs:
+  integration-test:
+    name: VSphere Integration Test
+    runs-on: self-hosted
+    timeout-minutes: 120
+    env: 
+      # These secrets are repository secrets
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+      B64_CREDS: ${{ secrets.B64_ENCODED_CI_CREDS }}
+      # GITHUB_TOKEN is an automatically included secret available to the workflow
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+      - name: Set cluster resources env
+        # the cluster resources are expected to be provided to pytest as an env var which holds a base64 encoded string
+        # since it will be a multiline env var special care is needed
+        # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+        run: |
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "B64_RESOURCES<<$EOF" >> "$GITHUB_ENV"
+          base64 test/data/ci-resources.yaml >> "$GITHUB_ENV"
+          echo "$EOF" >> "$GITHUB_ENV"
```

Comment:
I suspect this could be simplified by using `base64 -w0` to make it a single line.